docs/pages/reference/config.mdx

```diff
@@ -38,6 +38,63 @@ enable you to roll back to the previous configuration if you need to.
 
 </Notice>
 
+## Configuration file versions
+
+Teleport's configuration file is versioned independently from Teleport's
+releases. Teleport clusters currently support all versions of the configuration
+file. 
+
+The latest version is `v3`. If your configuration file has a lower version, we
+recommend moving to `v3` when possible for the best Teleport experience.
+
+<Notice type="danger">
+
+If a configuration file does not include a `version` field, the `teleport`
+binary reading the configuration assumes it is `v1`.
+
+</Notice>
+
+Here are the features that require users to opt in by explicitly upgrading their
+configuration to a newer version.
+
+### TLS Routing
```

Comment:
I'm not sure this is better, TBH. My goal was to help with questions like "whats the difference between config v2 and config v3" - that information seems harder to find now that it's hidden behind a section called "TLS Routing."

Additionally, I believe TLS routing works fine with config v1, config v2 is used to opt out of running the legacy listeners.